targets/TARGET_STM/TARGET_STM32F0/mbed_crc_api.c

```diff
@@ -0,0 +1,69 @@
+#include "crc_api.h"
+
+#include "platform/mbed_assert.h"
+
+#ifdef DEVICE_CRC
+
+#include "stm32f0xx_hal.h"
+
+static CRC_HandleTypeDef current_state;
+static uint32_t final_xor;
+
+bool hal_crc_is_supported(const crc_mbed_config_t* config)
+{
+  if (config == NULL) {
+    return false;
+  }
+
+  if (config->polynomial != DEFAULT_CRC32_POLY) {
```

Comment:
Shouldn't we use HAL definitions if possible. It looks like DEFAULT_CRC32_POLY corresponds POLY_32BIT_ANSI.
Please note that while testing the CRC HAL API I can not use target specific definitions. So I think that it is a good assumption that at least one of the predefined polynomials must be supported.